pyslim/slim_metadata.py

```diff
@@ -11,6 +13,431 @@
+                                "binaryFormat": "i",
+                                "index": 1
+                            },
+                            "selection_coeff": {
+                                "type": "number",
+                                "description": "This mutation's selection coefficient.",
+                                "binaryFormat": "f",
+                                "index": 2
+                            },
+                            "subpopulation": {
+                                "type": "integer",
+                                "description": "The ID of the subpopulation this mutation occurred in.",
+                                "binaryFormat": "i",
+                                "index": 3
+                            },
+                            "slim_time": {
+                                "type": "integer",
+                                "description": "The SLiM generation counter when this mutation occurred.",
+                                "binaryFormat": "i",
+                                "index": 4
+                            },
+                            "nucleotide": {
+                                "type": "integer",
+                                "description": "The nucleotide for this mutation (0=A , 1=C , 2=G, 3=T, or -1 for none)",
+                                "binaryFormat": "b",
+                                "index": 5
+                            }
+                        },
+                        "required": ["mutation_type", "selection_coeff", "subpopulation",
+                                     "slim_time", "nucleotide"],
+                        "additionalProperties": False
+                    }
+                }
+            },
+            "required": ["mutation_list"],
+            "additionalProperties": False,
+            "examples": [
+                {
+                 "mutation_list" : [ {
+                     "mutation_type" : 1,
+                     "selection_coeff" : -0.2,
+                     "subpopulation" : 0,
+                     "slim_time" : 243,
+                     "nucleotide" : 3
+                 } ]
+                }
+            ]
+        },
+        "node" : 
+        {
+            "$schema": "http://json-schema.org/schema#",
+            "description": "SLiM schema for node metadata.",
+            "codec": "struct",
+            "type": ["object", "null"],
+            "properties": {
+                "slim_id": {
+                    "type": "integer",
+                    "description": "The 'pedigree ID' of this chromosome in SLiM.",
+                    "binaryFormat": "q",
+                    "index": 0},
+                "is_null": {
+                    "type": "integer",
+                    "description": "Whether this node describes a 'null' (non-existant) chromosome.",
+                    "binaryFormat": "B",
```

Comment:
How come this isn't a `?` (`bool`). The example below suggests it is.

Reply:
That's because in [the SLiM code](https://github.com/petrelharp/SLiM/blob/tskit_update/core/slim_sim.h#L135) they're defined as `uint8_t`. Maybe this would all work out fine and be equivalent? I don't know?

Reply:
but, they *are* boolean, yes.

Reply:
I think `bool` size is compiler-dependant in C++ so SLiM needs to be specify `uint8_t`. In python `?` will always be one-byte. Should all work out fine.

Reply:
Ok - so I'll leave it as-is (right?).

Reply:
I think @benjeffery was saying we need to use `?` in the schema, and make sure that the corresponding value in SLiM is always a `uint8_t`; we should never used `b` or `B` in the schema, and we should never use `bool` in the corresponding C++ code.  Do I have that right?

Reply:
Yes, ? In python uint_8 in C++. Sorry I wasn't clearer!

Reply:
Oh, I got it. =)

Reply:
fixed in 151875a698